redhat_access_insights/connection.py

```diff
@@ -169,6 +153,13 @@ def get_proxies(self, config):
                 conf_proxy = scheme + location
             logger.debug("CONF Proxy: %s", conf_proxy)
             proxies = {"https": conf_proxy}
+
+        if os.environ.get('NO_PROXY') and conf_proxy:
+            logger.debug("Invalid Proxy configuration. "
+                         "You have environment variable NO_PROXY set "
+                         "as well as 'proxy' set in your configuration file. "
+                         "This is invalid. Do not expect normal results.")
```

Comment:
I don't think we need to have a proxy setting in the conf file at all. It should be handled via HTTPS_PROXY and NO_PROXY. This would ensure that everything is handled transparently behind the scenes by urllib/requests and the OS level, therefore we need not worry about any logic on our end. Although, the logic still has to be in there now at this point to maintain backwards compatibility, however moving foward we can say we do not support proxy config file variables and we only support HTTPS_PROXY or NO_PROXY environment variable configurations. We certainly don't want to start setting environment variables in our code.